content/developer/reference/guidelines.rst

```diff
@@ -999,17 +999,88 @@ More precise JS guidelines are detailed in the `github wiki  <https://github.com
+
+- Each block element should be on its own line.
+- Small inline/style/text elements can be placed either on the same line as their parent or on their own line (depending on their importance and size).
+- Indentation of 4 spaces is applied between a parent and its children elements. The closing tag of a parent should always be at the same indentation level as its opening tag.
+- It is preferable to use instructions (such as `t-if`) directly on the element on which they apply (as opposed to having them on a separate `<t>` element).
+
+Naming
+~~~~~~
+
+The root element and each other *important* element of the template should have a `name` attribute to ease the readability of the template.
+
+.. note::
+    The importance of an element in regard to this rule is left to the appreciation of the reviewer. If the structure of the template is hard to follow when reading just the first words of each line, it is probably a sign that some `name` attributes are missing on those lines.
+
+The name on the root element should follow the format `o-ComponentName`. And for children elements `o-ComponentName-partName` where:
+
+- `o-` is a must have Odoo prefix.
+- `ComponentName` is the PascalCase name of the current component/template.
+- `-` is a mandatory hyphen to separate the component name and the part name.
+- `partName` is the camelCase name of the child element referred by this `name` attribute.
+
+Attributes
+~~~~~~~~~~
+
+All the attributes of an element should be on the same line as the element itself. No character limit per line.
+
+.. note::
+    For extremely long lines, it is left to the appreciation of the reviewer to accept multi-line attributes within an element (but it is not mandatory).
+
+    In that case each attribute should be on its own line, with an indentation of 4 spaces relative to the element itself. The closing of the element (either `>` or `/>`) should be on its own line as well, with the same indentation as the corresponding opening character.
+
+The following order is applied for attributes:
+
+ #. `name`
+ #. `t-if`, `t-elif`, `t-else`
+ #. `t-foreach`, `t-as`
+ #. attributes specific to this type of element (eg. `type` for `<input>`, `src` for `<img>`)
+ #. `t-on-`
+ #. `class`
+ #. other generic attributes (eg. `title`, `aria-`)
+ #. `id`
+ #. OWL technical attributes (eg. `t-key`, `t-ref`)
+
+`t-att-` and `t-attf-` variations are ordered just after their static counter-part, in this order respectively.
+
+.. note:: `id` should not be used unless absolutely necessary (such as for making an anchor, or as the `for` target of a `label`), and then it should be uniquely generated.
+
+Style
+~~~~~
+
+Prefer using Bootstrap components and utilities over anything else, by applying their classes on the template. Custom style should only be used if absolutely necessary.
+
+Inside the `class` attribute itself, the following order should be respected:
+
+ #. Custom template class (only if necessary for adding custom style).
```

Comment:
same as above. this is way too much. This does not make a template easier to review. It will make life for everyone harder without a good reason. This should be just a suggestion, but nothing more

Reply:
Same as above too, ideally this should be automated and we would never have to worry about it.

Reply:
same as above obviously :)